File: archive/speedgrader/app/src/main/java/com/instructure/speedgrader/fragments/PdfSubmissionFragment.java
/*
 * Copyright (C) 2016 - present  Instructure, Inc.
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, version 3 of the License.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */

package com.instructure.speedgrader.fragments;

import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.instructure.annotations_library.FetchFileAsyncTask.FetchFileCallback;
import com.instructure.pandautils.utils.Const;
import com.instructure.speedgrader.R;
import com.instructure.speedgrader.util.App;
import com.instructure.speedgrader.util.CanvasPDFCache;
import com.pspdfkit.ui.PSPDFFragment;

import java.io.File;

public class PdfSubmissionFragment extends BaseSubmissionView {

    private static final String LOG_TAG = "SG.Submission";

    private Fragment mFragment;

    private Uri mTempFileUri;

    @Override
    public int getRootLayout() {
        return R.layout.fragment_pdf_container;
    }

    @Override
    public boolean retainInstanceState() {
        return false;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = super.onCreateView(inflater, container, savedInstanceState);

        mFragment = getChildFragmentManager().findFragmentById(R.id.pdfContainer);
        if (mFragment == null) {
            if(mTempFileUri == null){
                CanvasPDFCache.getInstance(getActivity()).getInputStream(getContext(), getAttachment().getUrl(), new FetchFileCallback() {
                    @Override
                    public void onFileLoaded(File file) {
                        if(getContext() == null){
                            return;
                        }
                        mTempFileUri = Uri.fromFile(file);
                        mFragment = createPDFFragment();
                        attachFragment();
                    }
                });
            }else{
                mFragment = createPDFFragment();
                attachFragment();
            }
        } else {
            //Rotation
            attachFragment();
        }

        return rootView;
    }

    @Nullable private Fragment createPDFFragment(){
        if(mTempFileUri == null){
            return null;
        }
        return PSPDFFragment.newInstance(mTempFileUri, ((App)getActivity().getApplication()).getConfig());
    }

    private void attachFragment() {
        getChildFragmentManager().beginTransaction()
                .replace(R.id.pdfContainer, mFragment)
                .commit();
        getChildFragmentManager().executePendingTransactions();
    }

    @Override
    public void onDetach() {
        super.onDetach();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
    }

    @Override
    public void getBundleData(Bundle bundle) {
        super.getBundleData(bundle);
        setAttachment(currentSubmission.getAttachments().get(0));
    }

    @Override
    public void onSaveInstanceState(Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putParcelable(Const.URI, mTempFileUri);
    }
}

File: apps/teacher/src/main/java/com/instructure/teacher/fragments/PdfSubmissionFragment.kt
/*
 * Copyright (C) 2025 - present Instructure, Inc.
 *
 *     This program is free software: you can redistribute it and/or modify
 *     it under the terms of the GNU General Public License as published by
 *     the Free Software Foundation, version 3 of the License.
 *
 *     This program is distributed in the hope that it will be useful,
 *     but WITHOUT ANY WARRANTY; without even the implied warranty of
 *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *     GNU General Public License for more details.
 *
 *     You should have received a copy of the GNU General Public License
 *     along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 */
package com.instructure.teacher.fragments

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.activityViewModels
import com.instructure.pandautils.base.BaseCanvasFragment
import com.instructure.pandautils.features.speedgrader.SpeedGraderSharedViewModel
import com.instructure.pandautils.utils.LongArg
import com.instructure.pandautils.utils.StringArg
import com.instructure.teacher.features.speedgrader.PdfTeacherSubmissionView

class PdfSubmissionFragment : BaseCanvasFragment() {

    private val sharedViewModel: SpeedGraderSharedViewModel by activityViewModels()

    private var pdfUrl by StringArg()
    private var courseId by LongArg()
    private var assigneeId by LongArg()

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        return PdfTeacherSubmissionView(requireActivity(), pdfUrl, courseId, assigneeId, childFragmentManager)
    }

    companion object {

        const val PDF_URL = "pdfUrl"
        const val COURSE_ID = "courseId"
        const val ASSIGNEE_ID = "assigneeId"

        fun newInstance(url: String, courseId: Long) = PdfSubmissionFragment().apply {
            pdfUrl = url
            this.courseId = courseId
        }

        fun createBundle(url: String, courseId: Long, assigneeId: Long): Bundle {
            return Bundle().apply {
                putString(PDF_URL, url)
                putLong(COURSE_ID, courseId)
                putLong(ASSIGNEE_ID, assigneeId)
            }
        }
    }
}